File: factorial/javascript_factorial/factorial.js
factorial = function( n ) {
  if( n == 0 ) {
    return 0;
  } else if( n == 1 ) {
    return 1;
  } else {
    return (n * factorial(n - 1));
  }
}

//document.write("factorial(5)=" + factorial(5));
for( var idx_i = 1; idx_i < 15; idx_i++ ) {
  console.log("factorial(" +idx_i + ")=" + factorial(idx_i));
}

File: factorial/clojure_factorial/src/factorial.clj
(ns factorial
  (:gen-class))

(defn factorial [x]
  "factorial"
   (loop [n x f 1]
     (if (= n 1)
       f
       (recur (dec n) (* f n)))))

(defn -main
  "entry_point"
  [& args]
  (loop [idx_i 1]
    (when (< idx_i 15)
      (println (format "factorial(%d)=<%d>" idx_i (factorial idx_i)))
      (recur (inc idx_i))
    )
  )
)
